src/commands/slow-mode.ts

```diff
@@ -0,0 +1,579 @@
+	await interaction.deferReply({ ephemeral: true });
+
+	const channel = interaction.options.getChannel('channel') ?? interaction.channel;
+	if (!channel) {
+		throw new Error('No channel given');
+	}
+
+	if (channel.type !== ChannelType.GuildText) {
+		throw new Error('Slow mode only applies to text channels');
+	}
+
+	const slowMode = await SlowMode.findOne({
+		where: {
+			channel_id: channel.id
+		},
+		include: 'stages'
+	});
+
+	if (!slowMode) {
+		throw new Error(`No slow mode set for <#${channel.id}>`);
+	}
+
+	const threshold = interaction.options.getNumber('threshold', true);
+
+	const stage = slowMode.stages?.find(stage => stage.threshold === threshold);
+	if (!stage) {
+		throw new Error(`No stage set at ${threshold} messages per minute`);
+	}
+
+	const oldLimit = stage.limit;
+	await slowMode.removeStage(stage);
+	await slowMode.reload({ include: 'stages' });
+
+	const auditLogEmbed = new EmbedBuilder()
+		.setColor(0xC0C0C0)
+		.setTitle('Event Type: _Auto Slow-Mode Threshold Unset_')
+		.setDescription('――――――――――――――――――――――――――――――――――')
+		.setFields(
+			{
+				name: 'User',
+				value: `<@${interaction.user.id}>`
+			},
+			{
+				name: 'Channel',
+				value: `<#${channel.id}>`
+			},
+			{
+				name: 'Old Threshold',
+				value: `Limit of 1 message every ${oldLimit} seconds above ${threshold} messages per minute`
+			}
+		).setFooter({
+			text: 'Pretendo Network',
+			iconURL: interaction.guild!.iconURL()!
+		});
+
+	if (slowMode.stages) {
+		const stagesMessage = slowMode.stages.sort((a, b) => a.threshold - b.threshold).map(stage => {
+			return `1 message every ${stage.limit}s above ${stage.threshold} messages per minute`;
+		}).join('\n');
+	
+		auditLogEmbed.addFields([
+			{
+				name: 'Configured stages',
+				value: stagesMessage
+			}
+		]);
+	}
+		
+	await sendEventLogMessage(interaction.guild!, channel.id, auditLogEmbed);
+
+	await interaction.followUp({ content: `Unset the limit at ${threshold} messages per minute on <#${channel.id}>` });
+}
+
+async function enableAutoSlowModeHandler(interaction: ChatInputCommandInteraction): Promise<void> {
+	await interaction.deferReply({ ephemeral: true });
+
+	const channel = interaction.options.getChannel('channel') ?? interaction.channel;
+	if (!channel) {
+		throw new Error('No channel given');
+	}
+
+	if (channel.type !== ChannelType.GuildText) {
+		throw new Error('Slow mode only applies to text channels');
+	}
+
+	let slowMode = await SlowMode.findOne({
+		where: {
+			channel_id: channel.id
+		},
+		include: 'stages'
+	});
+
+	if (slowMode) {
+		slowMode.enabled = true;
+	} else {
+		slowMode = await SlowMode.create({
+			channel_id: channel.id,
+			window: 60000,
+			enabled: true
+		});
+	}
+
+	const window = interaction.options.getNumber('window');
+	if (window) {
+		slowMode.window = window;
+	}
+
+	await slowMode.save();
+
+	// * This returns a Promise but is specifically not awaited as it should spawn its own loop
+	handleSlowMode(interaction.guild!, slowMode);
```

Comment:
This has the potential to start a large number of `handleSlowMode` loops if the enable command is run multiple times for the same channel. I don't know if this could cause problems, but either way it seems reasonable to check if slowmode is already enabled first before re-enabling.